core/contracts/financial_templates/Liquidatable.sol

```diff
@@ -170,40 +169,40 @@ contract Liquidatable is PricelessPositionManager {
      *
      * Returns UUID of new liquidation for the sponsor
      */
-    function createLiquidation(address sponsor) public returns (uint lastIndexUsed) {
+    function createLiquidation(address sponsor) public returns (uint uuid) {
         // Attempt to retrieve Position data for sponsor
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
 
-        // Allocate space for new liquidation and increment index
-        lastIndexUsed = sponsorLiquidationIndex[sponsor];
-        LiquidationData storage newLiquidation = liquidations[sponsor][lastIndexUsed];
-        sponsorLiquidationIndex[sponsor] = (lastIndexUsed + 1);
-
-        // Read position data into liquidation
-        newLiquidation.tokensOutstanding = positionToLiquidate.tokensOutstanding;
-        newLiquidation.lockedCollateral = positionToLiquidate.collateral;
-        newLiquidation.liquidatedCollateral = positionToLiquidate.collateral.sub(
-            positionToLiquidate.withdrawalRequestAmount
+        // Construct liquidation object.
+        // Note: all dispute-related values are just zeroed out until a dispute occurs.
+        uint newLength = liquidations[sponsor].push(
+            LiquidationData({
+                expiry: getCurrentTime() + liquidationLiveness,
+                liquidator: msg.sender,
+                state: Status.PreDispute,
+                lockedCollateral: positionToLiquidate.collateral,
+                tokensOutstanding: positionToLiquidate.tokensOutstanding,
+                liquidatedCollateral: positionToLiquidate.collateral.sub(positionToLiquidate.withdrawalRequestAmount),
+                disputer: address(0),
+                disputeTime: 0,
+                settlementPrice: FixedPoint.fromUnscaledUint(0)
+            })
         );
 
-        // TODO: Should "destroy" the position somehow, rendering its create/redeem/deposit/withdraw methods uncallable
-        // This should reduce totalTokensOutstanding and lockedCollateral, and also withdrawal request amount?
-        positionToLiquidate.tokensOutstanding = FixedPoint.fromUnscaledUint(0);
-        positionToLiquidate.collateral = FixedPoint.fromUnscaledUint(0);
-
-        // Set parameters for new liquidation
-        newLiquidation.expiry = getCurrentTime() + liquidationLiveness;
-        newLiquidation.liquidator = msg.sender;
-        newLiquidation.state = Status.PreDispute;
+        // UUID is the index of the LiquidationData that was just pushed into the array, which is length - 1.
+        uuid = newLength.sub(1);
```

Comment:
Perhaps you comment on this above from @ptare 's comment, but have you empirically verified that `delete` does not shift around the array such that the `uuid`'s would have to change as well?

Reply:
I just verified this behavior in Remix -- delete on an element does not modify the length or any other element in the array (including other elements' positions).